perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -501,278 +433,245 @@ def _DistributionToBackendFormat(dist):
+        '--storage_provider=%s' % self.storage
+    ] + args + self.service.APIScriptArgs()
+    if FLAGS.object_storage_storage_class is not None:
+      cmd_parts += ['--object_storage_class',
+                    FLAGS.object_storage_storage_class]
+
+    return ' '.join(cmd_parts)
+
+# The benchmark functions below all have the same arguments:
+# Args:
+#   results: The results array to append to.
+#   metadata: a dictionary of metadata to add to samples.
+#   vm: The vm being used to run the benchmark.
+#   command_builder: an APIScriptCommandBuilder.
+#   service: an ObjectStorageService.
+#   bucket_name: The primary bucket to benchmark.
+#   regional_bucket_name: The secondary bucket to benchmark. Only
+#     used by the api_data scenario.
+
+# Raises:
+#   ValueError: unexpected test outcome is found from the API test script.
+
+
+def OneByteRWBenchmark(results, metadata, vm, command_builder,
+                       service, bucket_name, regional_bucket_name):
+  buckets = [bucket_name]
+  if regional_bucket_name is not None:
+    buckets.append(regional_bucket_name)
+
+  for bucket in buckets:
+    one_byte_rw_cmd = command_builder.BuildCommand([
+        '--bucket=%s' % bucket,
+        '--scenario=OneByteRW'])
+
+    _, raw_result = vm.RemoteCommand(one_byte_rw_cmd)
+    logging.info('OneByteRW raw result is %s', raw_result)
+
+    for up_and_down in ['upload', 'download']:
+      search_string = 'One byte %s - (.*)' % up_and_down
+      result_string = re.findall(search_string, raw_result)
+      sample_name = ONE_BYTE_LATENCY % up_and_down
+      if bucket == regional_bucket_name:
+        sample_name = 'regional %s' % sample_name
+
+      if len(result_string) > 0:
+        _JsonStringToPercentileResults(results,
+                                       result_string[0],
+                                       sample_name,
+                                       LATENCY_UNIT,
+                                       metadata)
       else:
-        logging.error('Exceeded max retry limit for removing the empty bucket')
-        raise BucketRemovalError('Failed to remove the bucket')
+        raise ValueError('Unexpected test outcome from OneByteRW api test: '
+                         '%s.' % raw_result)
+
+
+def SingleStreamThroughputBenchmark(results, metadata, vm, command_builder,
+                                    service, bucket_name, regional_bucket_name):
+  single_stream_throughput_cmd = command_builder.BuildCommand([
+      '--bucket=%s' % bucket_name,
+      '--scenario=SingleStreamThroughput'])
+
+  _, raw_result = vm.RemoteCommand(single_stream_throughput_cmd)
+  logging.info('SingleStreamThroughput raw result is %s', raw_result)
+
+  for up_and_down in ['upload', 'download']:
+    search_string = 'Single stream %s throughput in Bps: (.*)' % up_and_down
+    result_string = re.findall(search_string, raw_result)
+    sample_name = SINGLE_STREAM_THROUGHPUT % up_and_down
+
+    if len(result_string) > 0:
```

Comment:
style nitpick: why not check first for the error condition  and raise the error immediately? That way you don't have to indent the for-loop into the if block.

I believe `re.findall()` will either return an empty list or non-empty list, you can use Python's coercion to booleans False and True, respectively. (e.g. `if not result_string:`)